oci/layout/oci_dest.go

```diff
@@ -218,33 +219,108 @@ func (d *ociImageDestination) PutManifest(m []byte) error {
 	}
 
 	if d.ref.image == "" {
-		return errors.Errorf("cannot save image with empyt image.ref.name")
+		return errors.Errorf("cannot save image with empty image.ref.name")
 	}
 
+	// Start filling out the descriptor for this entry
+	desc := imgspecv1.Descriptor{}
+	desc.Digest = digest
+	desc.Size = int64(len(m))
+
+	// Only a "main" manifest gets the image name attached to it.
 	annotations := make(map[string]string)
-	annotations["org.opencontainers.image.ref.name"] = d.ref.image
+	if instanceDigest == nil {
+		annotations[imgspecv1.AnnotationRefName] = d.ref.image
+	}
 	desc.Annotations = annotations
-	desc.Platform = &imgspecv1.Platform{
-		Architecture: runtime.GOARCH,
-		OS:           runtime.GOOS,
+
+	// If we knew the MIME type, we wouldn't have to guess here.
+	mt := manifest.GuessMIMEType(m)
+	switch mt {
+	case imgspecv1.MediaTypeImageManifest:
+		// Continue filling out the descriptor.
+		desc.MediaType = imgspecv1.MediaTypeImageManifest
+		// Try to figure out what the Platform field should contain.
```

Comment:
Now that I think about it more, `ChooseInstance()` doesn't care about platform information in the layout's internal index, since it's searching the primary manifest (which is an index), which has the information.  So yeah, dropped that.